L1Trigger/CSCTriggerPrimitives/src/CSCCathodeLCTProcessor.cc

```diff
@@ -1256,13 +1250,51 @@ void CSCCathodeLCTProcessor::encodeHighMultiplicityBits(
   std::vector<unsigned> station_thresholds = {
       thresholds_[csc_idx * 3], thresholds_[csc_idx * 3 + 1], thresholds_[csc_idx * 3 + 2]};
 
-  // assign the bits
-  for (unsigned i = 0; i < station_thresholds.size(); i++) {
-    if (hitsInTime >= station_thresholds[i]) {
-      inTimeHMT_ = i + 1;
+  //hard coded dead time as 2Bx, since showerNumTBins = 3, like firmware
+  // for example, nhits = 0 at bx7; = 100 at bx8; = 0 at bx9
+  //cathode HMT must be triggered at bx8, not bx7 and bx9
+  //meanwhile we forced 2BX dead time after active shower trigger
+  unsigned int deadtime =
+      showerNumTBins_ - 1;  // firmware hard coded dead time as 2Bx, since showerNumTBins = 3 in firmware
+  unsigned int dead_count = 0;
+  bool dead_status = false;
+
+  for (unsigned bx = 0; bx < CSCConstants::MAX_CLCT_TBINS; bx++) {
+    unsigned minbx = bx >= showerNumTBins_ / 2 ? bx - showerNumTBins_ / 2 : bx;
+    unsigned maxbx = bx < CSCConstants::MAX_CLCT_TBINS - showerNumTBins_ / 2 ? bx + showerNumTBins_ / 2
+                                                                             : CSCConstants::MAX_CLCT_TBINS - 1;
+    unsigned this_hitsInTime = 0;
+    bool isPeak = true;  //check whether total hits in bx is peak of nhits over time bins
+    /*following is to count number of hits over [minbx, maxbx], showerNumTBins=3 =>[n-1, n+1]*/
+    for (unsigned mbx = minbx; mbx <= maxbx; mbx++) {
+      this_hitsInTime += hitsInTime[mbx];
     }
-  }
 
-  // create a new object
-  shower_ = CSCShowerDigi(inTimeHMT_, false, theTrigChamber);
+    if (peakCheck_ and bx < CSCConstants::MAX_CLCT_TBINS - showerNumTBins_ / 2 - 1) {
+      if (hitsInTime[minbx] < hitsInTime[maxbx + 1] or
+          (hitsInTime[minbx] == hitsInTime[maxbx + 1] and hitsInTime[bx] < hitsInTime[bx + 1]))
+        isPeak = false;  //next bx would have more hits or in the center
+    }
+    if (dead_count > 0) {
+      dead_count--;
+      dead_status = true;
+    } else
+      dead_status = false;
+
+    unsigned this_inTimeHMT = 0;
+    // require at least nLayersWithHits for the central time bin
+    // do nothing if there are not enough layers with hits
+    if (layersWithHits[bx].size() >= minLayersCentralTBin_ and !dead_status and isPeak) {
+      // assign the bits
+      for (unsigned i = 0; i < station_thresholds.size(); i++) {
+        if (this_hitsInTime >= station_thresholds[i]) {
+          this_inTimeHMT = i + 1;
+          dead_count = deadtime;
+        }
+      }
```

Comment:
If this has to be overwritten at every station_threshold, it is probably more convenient:
```suggestion
      if (station_thresholds.size() > 0) {
        for (unsigned i = station_thresholds.size() - 1; i >= 0; i--) {
          if (this_hitsInTime >= station_thresholds[i]) {
            this_inTimeHMT = i + 1;
            dead_count = deadtime;
            break;
          }
        }
      }
```

Reply:
Hi @perrotta 
`this_inTimeHMT` needs to count how many `station_thresholds`(loose, nominal, tight) it passes. It is used to set the bit for the `CSCShowerDigi` object [here](https://github.com/cms-sw/cmssw/pull/38941/files#diff-fa43c5c8243291071f6787aa0b1586f9244ece75c3f67547023b0e81ab7431caR12)

So we can't have the `break` and the original implementation seems more natural to me.

I'm also addressing the other comments.

Reply:
@kakwok but here you are not counting how many station_thresholds it passes, just the index +1 of the last station over threshold. Which is the first one encountered if you loop in the reverse order, as I suggested...

Reply:
@perrotta I see what you mean now. I'm fine with counting in reverse order then.

Reply:
fixed as suggested.